doc/services/logging/cs_stm.rst

```diff
@@ -0,0 +1,241 @@
+.. _logging_cs_stm:
+
+Multi-domain logging using ARM Coresight STM
+############################################
+
+The Arm CoreSight SoC-400 is a comprehensive library of components for the creation of debug and
+trace functionality within a system. The STM (System Trace Macrocell) is a trace source that
+is integrated into a CoreSight system, designed primarily for high-bandwidth trace of
+instrumentation embedded into software. This instrumentation is made up of memory-mapped writes to
+the STMESP (STM Extended Stimulus Port) peripheral registers. Multiple cores can share and directly
+access the STM without being aware of each other. Each core has 65536 stimulus ports (identical
+register sets) which can be independently accessed thus there is no need for locking when various
+contexts want to log the data.
+
+STM Extended Stimulus Port (STMESP)
+***********************************
+
+Local domain (each core) has access to its own set of STMESP peripherals. Each set has registers for writing data (with or without timestamp and with or without marking) and writing a flag. Each write
+to STMESP register results in the user data being encoded into MIPI System Trace Protocol v2 (STPv2).
+Hardware manages multiplexing of data from different STMESP register sets and from different cores
+by adding Master and Channel opcodes into the stream when necessary. Timestamp (from common source for all cores) and synchronization opcodes are autonomously added by the STM.
+
+Data stream generated by STM can be directly send to TPIU (Trace Port Interface Unit) or saved in
```

Comment:
To be precise, data can be sent to any trace sink. Another option is ETB. This document should not limit the options to TPIU and ETR. Just show them as examples.

Reply:
rephrased a bit.